clang/lib/Sema/SemaType.cpp

```diff
@@ -7024,13 +7024,12 @@ static bool handleFunctionTypeAttr(TypeProcessingState &state, ParsedAttr &attr,
                << (int)Sema::CallingConventionIgnoredReason::VariadicFunction;
 
       attr.setInvalid();
-      if (S.getLangOpts().SYCLIsDevice) {
-        S.SYCLDiagIfDeviceCode(attr.getLoc(), diag::err_cconv_varargs)
-          << FunctionType::getNameForCallConv(CC);
-        return true;
-      } else
-      return S.Diag(attr.getLoc(), diag::err_cconv_varargs)
-             << FunctionType::getNameForCallConv(CC);
+      // Disable this diagnostic for SYCL
+      // Note that deferring does not work since the declaration
+      // need not be in a function context.
```

Comment:
I guess SYCL Compiler should skip the diagnostic for declaration, but emit an error if vararg function is called. Right?

Reply:
Yes.  I wanted to do the latter as a separate patch, as currently this is preventing the compilation of common header files.